src/lib/useEthereumTransaction.js

```diff
@@ -174,7 +173,6 @@ export default function useEthereumTransaction(
   ]);
 
   const reset = useCallback(() => {
-    setUnsignedTransaction(undefined);
```

Comment:
Why are we no longer clearing this on-reset?

Reply:
And should we maybe clear `nonce` here as well, so we fetch a fresh one? Very edge case, but can't hurt.

Reply:
good call on nonce

clearing the unsigned transaction put us in a weird state (especially for components that did something like `useLifecycle(() => construct(...))` because after a reset there was no unsigned transaction and lifecycle wouldn't run again.